spikeinterface/preprocessing/tests/test_highpass_spatial_filter.py

```diff
@@ -0,0 +1,190 @@
+import spikeinterface as si
+import spikeinterface.preprocessing as spre
+import spikeinterface.extractors as se
+import pytest
+import numpy as np
+from copy import deepcopy
+
+from spikeinterface.core.testing_tools import generate_recording
+import spikeinterface.widgets as sw
+
+try:
+    import spikeglx
+    import neurodsp.voltage as voltage
+    HAVE_IBL_NPIX = True
+except ImportError:
+    HAVE_IBL_NPIX = False
+
+DEBUG = False
+if DEBUG:
+    import matplotlib.pyplot as plt
+    plt.ion()
+    plt.show()
+
+# ----------------------------------------------------------------------------------------------------------------------
+# Tests
+# ----------------------------------------------------------------------------------------------------------------------
+
+@pytest.mark.parametrize("lagc", ["ibl", None, 1, 150])
+def test_highpass_spatial_filter_real_data(lagc):
+    """
+    Test highpass spatial filter IBL vs. SI implimentations. Download
+    a real dataset from SI repo, run both pipelines and check results.
+
+    Requires a bit of exchanging argument types for differences in inputs
+    between IBL and SI.
+
+    Cycle through the lagc (automatic gain control) arguments (default
+    is IBL default params, False and None are no lagc. Do this here rather
+    than in larger test below to avoid unecessary tests, of every
+    combination with False and None.
+
+    Argument changes:
+        - lagc is changed to a dictionary, so that sampling_interval
+          can be easily passed to IBL helper function, and so that
+          window_length can be specified in seconds.
+
+        - ntr_pad can be None for SI, lagc can be "default", None or False.
+
+    use DEBUG = true to visualise.
+
+    """
+    options = dict(lagc=lagc, ntr_pad=25, ntr_tap=50, butter_kwargs=None)
+    print(options)
+
+    ibl_data, si_recording = get_ibl_si_data()
```

Comment:
I would not do this on gihut actions. It will slow down even more.
We can keep this test locally but skip if on github.